src/seedu/addressbook/AddressBook.java

```diff
@@ -359,6 +368,61 @@ public static String executeCommand(String userInputString) {
     }
 
     /**
+     * Edits person identified using last displayed index.
+     *
+     * @param commandArgs full command args string from the user
+     * @return feedback display message for the operation result
+     */
+    private static String executeEditPerson(String commandArgs) {
+    	// try decoding parameters from the raw args
+        final Optional<ArrayList<String>> decodeResult = decodeEditDataFromString(commandArgs);
+
+        // checks if args are valid (decode result will not be present if the person is invalid)
+        if (!decodeResult.isPresent()) {
+            return getMessageForInvalidCommandInput(COMMAND_EDIT_WORD, getUsageInfoForEditCommand());
+        }
+    	
+        final int targetVisibleIndex = Integer.parseInt(decodeResult.get().get(0));
+        if (!isDisplayIndexValidForLastPersonListingView(targetVisibleIndex)) {
+            return MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+        }
+        else {
+        	editPersonFromAddressBook(decodeResult.get(), targetVisibleIndex);
+        	return String.format(MESSAGE_EDIT_PERSON_SUCCESS,
+        					getMessageForFormattedPersonData(latestPersonListingView.get(targetVisibleIndex - DISPLAYED_INDEX_OFFSET)));
+        }
+    }
+
+    /**
+     * Displays all persons in the address book to the user; in alphabetical order.
+     *
+     * @return feedback display message for the operation result
+     */
+    private static boolean editPersonFromAddressBook(ArrayList<String> editData, int indexToBeEdited){
+    	Person toEdit = latestPersonListingView.get(indexToBeEdited - DISPLAYED_INDEX_OFFSET);
+    	if (editData.get(PERSON_EDIT_INDEX_NAME) != null)
+    		toEdit.setName(editData.get(PERSON_EDIT_INDEX_NAME));
+    	if (editData.get(PERSON_EDIT_INDEX_PHONE) != null)
+    		toEdit.setPhone(editData.get(PERSON_EDIT_INDEX_PHONE));
+    	if (editData.get(PERSON_EDIT_INDEX_EMAIL) != null)
+    		toEdit.setEmail(editData.get(PERSON_EDIT_INDEX_EMAIL));
+    	savePersonsToFile(getAllPersonsInAddressBook(), storageFilePath);
+    	return true;
+    }
+    
+	/**
+     * Displays all persons in the address book to the user; in alphabetical order.
+     *
+     * @return feedback display message for the operation result
+     */
+	private static String executeSortAndListAllPersonsInAddressBook() {
+    	ArrayList<Person> toBeDisplayed = getAllPersonsInAddressBook();
+    	Collections.sort(toBeDisplayed);
+    	showToUser((toBeDisplayed));
+        return getMessageForPersonsDisplayedSummary(toBeDisplayed);
+	}
```

Comment:
Indentation problem here?. Use spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.